File: BookingApp/src/main/java/com/example/BookingApp/reservations/model/RevisionReply.java
package com.example.BookingApp.reservations.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;

@Entity
@Table(name="replies")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class RevisionReply {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator="seq")
    @GenericGenerator(name = "seq", strategy="increment")
    private Long id;
    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    private Revision revision;
    private Long ownerId;
    private String description;
    private boolean approved;
}

File: BookingApp/frontend/src/app/shared/models/RevisionReply.ts
import { Revision } from "./Revision";

export class RevisionReply{
    public id : Number;
    public revision : Revision;
    public ownerId : Number;
    public description : String;

    constructor(id : Number,revision : Revision, ownerId : Number,description : String){
        this.id = id;
        this.revision = revision;
        this.ownerId = ownerId;
        this.description = description;
    }
}
